src/components/DeploymentForm.vue

```diff
@@ -119,14 +134,21 @@
   const { value: workPoolName } = useField<string | null>('workPoolName')
   const { value: workQueueName } = useField<string | null>('workQueueName')
   const { value: tags } = useField<string[] | null>('tags')
+  const { value: infrastructureOverrides } = useField<string>('infrastructureOverrides')
 
   const emit = defineEmits<{
     (event: 'submit', value: DeploymentUpdate): void,
     (event: 'cancel'): void,
   }>()
 
   const submit = handleSubmit((values) => {
-    emit('submit', values)
+    try {
+      JSON.parse(infrastructureOverrides.value)
+      emit('submit', values)
+    } catch (error) {
+      showToast(localization.error.invalidJSON, 'error')
+      throw new Error('Invalid JSON')
+    }
```

Comment:
Instead of this bespoke validation, we have an `isValidJson` validation utility you can pass to the field which will prevent form submission